### DISCUSSION TITLE
Link Preview works from Quarto Pub but not from Github (docs)

### DISCUSSION THREAD

**Discussion by ariewjy** [quarto-dev/quarto-cli#6148] [Q&A] [answered]
### Link Preview Does not Work if coming from Github, but does in Quarto Pub

I am using quarto to create my personal website, but somehow the link preview only works when it was pushed to quarto pub, but not if pushed to github. 

Here is link for quarto pub and github

quarto-pub: https://ariewjy.quarto.pub/geoscientist-who-codes/posts/007-quarto-python-tutorial/
<img width="1141" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/73975333/d4d827a9-1a12-4bfd-8de6-a791ed7630c3">

github: https://ariewjy.github.io/posts/007-quarto-python-tutorial/
![CleanShot 2023-07-08 at 17 04 49](https://github.com/quarto-dev/quarto-cli/assets/73975333/5e32a5d3-178d-4b11-b2b4-b6654a1afe70)

Here is the link of my personal website repo: https://github.com/ariewjy/ariewjy.github.io

Does anyone have similar experience or have any idea on what went wrong?

Many thanks
ARie

**Comment by cderv** [comment, marked as answer]
The link to OG in the website deployed on `github.io` is 
````html
<meta property="og:image" content="ariewjy.github.io/posts/007-quarto-python-tutorial/images/image0.jpg">
````

This will be interpreted in your tool https://www.opengraph.xyz/url/https%3A%2F%2Fariewjy.github.io%2Fposts%2F007-quarto-python-tutorial%2F  by this wrong url (see on the left of you screenshot the image link parsed)
```
https://ariewjy.github.ioariewjy.github.io/posts/007-quarto-python-tutorial/images/image0.jpg
````

I guess we build the link based on 
````yaml
  site-url: ariewjy.github.io
  site-path: "https://ariewjy.github.io/"
````

From our doc note that you don't need `site-path` unless you have a [non root site ](https://quarto.org/docs/websites/website-navigation.html#non-root-site-paths)

So just change it to 

````yaml
  site-url: https://ariewjy.github.io
````

and it should solve your issu. See example at Social metadata doc: https://quarto.org/docs/websites/website-tools.html#social-metadata

**Reply by ariewjy** [reply]
This works wonder, thanks for the guidance and I learned something new about html.
